src/project-converter.ts

```diff
@@ -87,21 +87,52 @@ export class ProjectConverter {
    * Check if a document is explicitly excluded or has already been converted
    * to decide if it should be converted or skipped.
    */
-  shouldConvertDocument(document: Document): boolean {
+  private _shouldConvertDocument(document: Document): boolean {
     const documentUrl = getDocumentUrl(document);
     return !this.conversionResults.has(documentUrl) &&
         !this.conversionSettings.excludes.has(documentUrl);
   }
 
+  /**
+   * Convert document dependencies. Should be called early in the conversion
+   * process so that it can read this document's dependencies' exports.
+   */
+  private _convertDependencies(
+      document: Document, visited: Set<OriginalDocumentUrl>) {
+    for (const htmlImport of DocumentConverter.getAllHtmlImports(document)) {
+      // Only convert what we need, ignore excluded/already converted documents.
+      if (!this._shouldConvertDocument(htmlImport.document)) {
```

Comment:
Can we remove this check? It looks like `this.convertDocumentToJs` also checks if it should convert, and early exits if not

Reply:
We get bad "Cycle in dependency graph" warnings below if this is removed. I'll add that info to the comment to make this more clear